core/drivers/gic.c

```diff
@@ -604,3 +605,67 @@ static void gic_op_set_affinity(struct itr_chip *chip, size_t it,
 
 	gic_it_set_cpu_mask(gd, it, cpu_mask);
 }
+
+#ifdef CFG_DT
+/* Callback for "interrupt-extended" GIC interrupts in consumer DT nodes */
+static TEE_Result dt_get_gic_chip_cb(struct dt_pargs *arg, void *priv_data,
+				     struct itr_desc **out_itr_desc)
+{
+	int itr_num = DT_INFO_INVALID_INTERRUPT;
+	struct itr_chip *chip = priv_data;
+	struct itr_desc *desc = NULL;
+	uint32_t type = 0;
+	uint32_t prio = 0;
+	int n = 0;
+
+	/*
+	 * gic_dt_get_irq() expects phandle arguments passed are still in DT
+	 * format (big-endian) whereas struct dt_pargs carries converted
+	 * formats. Therefore swap again phandle arguments.
+	 */
+	for (n = 0; n < arg->args_count; n++)
+		arg->args[n] = cpu_to_fdt32(arg->args[n]);
```

Comment:
This is not nice. It modifies `arg` content. If called several times (some late init deferral), them `arg` content will not be reliable. I'll change into using a local copy of the arguments on which to call `cpu_to_fdt32()`.